traffic_ops/traffic_ops_golang/cachegroup/queueupdate.go

```diff
@@ -49,10 +49,14 @@ func QueueUpdates(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	if reqObj.CDN == nil && reqObj.CDNID == nil {
-		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("cdn does not exist"), nil)
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("cdn is a required field"), nil)
 		return
 	}
-	if reqObj.CDN == nil || *reqObj.CDN == "" {
+	if *reqObj.CDN == "" {
```

Comment:
This doesn't seem to make sense, with the check below. This will panic if `reqobj.CDN` is nil, but `reqobj.CDNID` is not nil. 

It seems like, with the check below, that this should just be removed. Right? If not, it still needs to check `reqObj.CDN == nil` to avoid panicing.

Reply:
Actually, I was mistaken: this does need to exist, even with the check below. Because below doesn't account for the case if `CDNID == nil && *reqobj.CDN == ""`. 

I think this should be removed, _and_ the above check `if reqObj.CDN == nil && reqObj.CDNID == nil {` should be changed to `if reqObj.CDNID == nil && (reqObj.CDN == nil || *reqObj.CDN == "" {`.

I think that handles all the cases, and IMO makes it obvious, "if the ID is nil _and_ the name is either nil or empty, return an error".